script.go

```diff
@@ -31,7 +31,7 @@ type Pipe struct {
 	stdout, stderr io.Writer
 	httpClient     *http.Client
 
-	// because pipe stages are concurrent, protect 'err'
+	// because pipe stages are concurrent, protect 'err' and 'stderr'
```

Comment:
Notice how the `err` field is listed below the `mu` field, in its own 'block', separated from the other fields by a blank line. This is a nice way to show, lexically, that the mutex is 'over' these fields. To maintain this, let's move the `stderr` field in underneath the mutex too.

Is there any reason not to include the `stdout` field too? I mean, if a race condition exists on `stderr`, doesn't it also exist on `stdout`?

Reply:
> Notice how the `err` field is listed below the `mu` field, in its own 'block', separated from the other fields by a blank line. This is a nice way to show, lexically, that the mutex is 'over' these fields. To maintain this, let's move the `stderr` field in underneath the mutex too.
addressed!
> Is there any reason not to include the `stdout` field too? I mean, if a race condition exists on `stderr`, doesn't it also exist on `stdout`?
The race condition exists on `stderr` because it is accessed inside a goroutine at https://github.com/bitfield/script/blob/master/script.go#L388-L390. The same is not true for `stdout` so I don't believe a race condition exists on `stdout`. We could still protect it with a mutex to match the behavior of `stdout` if we'd like however. Let me know what you think.